test/helpers/kubectl.go

```diff
@@ -829,16 +831,50 @@ func (kub *Kubectl) CiliumInstall(dsPatchName, cmPatchName string) error {
 		return err
 	}
 
-	if err := deployPatch(cmPathname, ManifestGet(cmPatchName)); err != nil {
+	if err := deployPatch(cmPathname, getK8sDescriptorPatch(cmPatchName)); err != nil {
 		return err
 	}
 
-	if err := deployPatch(dsPathname, ManifestGet(dsPatchName)); err != nil {
+	if err := deployPatch(dsPathname, getK8sDescriptorPatch(dsPatchName)); err != nil {
 		return err
 	}
 	return nil
 }
 
+// CiliumInstall installs all Cilium descriptors into kubernetes.
+// dsPatchName corresponds to the DaemonSet patch that will be applied to the
+// original Cilium DaemonSet descriptor.
+// cmPatchName corresponds to the ConfigMap patch that will be applied to the
+// original Cilium ConfigMap descriptor.
+// Returns an error if any patch or if any original descriptors files were not
+// found.
+func (kub *Kubectl) CiliumInstall(dsPatchName, cmPatchName string) error {
+	return kub.ciliumInstall(dsPatchName, cmPatchName, GetK8sDescriptor, ManifestGet)
+}
+
+// CiliumInstallVersion installs all Cilium descriptors into kubernetes for
+// a given Cilium Version tag.
+// dsPatchName corresponds to the DaemonSet patch that will be applied to the
+// original Cilium DaemonSet descriptor of that given Cilium Version tag.
+// cmPatchName corresponds to the ConfigMap patch that will be applied to the
+// original Cilium ConfigMap descriptor of that given Cilium Version tag.
+// Returns an error if any patch or if any original descriptors files were not
+// found.
+func (kub *Kubectl) CiliumInstallVersion(dsPatchName, cmPatchName, versionTag string) error {
+	getK8sDescriptorPatch := func(filename string) string {
+		fullPath := filepath.Join(manifestsPath, GetCurrentK8SEnv(), versionTag, filename)
+		_, err := os.Stat(fullPath)
+		if err == nil {
+			return filepath.Join(BasePath, fullPath)
+		}
+		return filepath.Join(BasePath, "k8sT", "manifests", filename)
+	}
+	getK8sDescriptor := func(filename string) string {
+		return fmt.Sprintf("https://raw.githubusercontent.com/cilium/cilium/%s/examples/kubernetes/%s/%s", versionTag, GetCurrentK8SEnv(), filename)
```

Comment:
I'm not sure, but I can't get why this is needed at all, could you explain?

Reply:
@eloycoto It is the official way we have access to the kubernetes descriptors for the `versionTag`.